Enhance ms teams functionalities part1 (#37268)
demisto/content: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md

3. Check the **long running instance** parameter in the integration instance configuration.
4. Set the **port** parameter. It's under the Connect section in the integration instance configuration.
5. If using the same bot from the XSOAR 6 instance, make sure to remove the bot from the team and to add it back:
    - Go to the Microsoft Teams app.
    - Go to your team, and click the three dots next to the name.
    - Go to **manage team** > **apps**.
    - Find your bot, and click the three dots in the same row.
    - Click **remove**.

>     - Add the bot to the team.

Reviewer: These steps should be numbered.  They are sequential steps.

3. Check the **long running instance** parameter in the integration instance configuration.
4. Set the **port** parameter. It's under the Connect section in the integration instance configuration.
5. If using the same bot from the XSOAR 6 instance, make sure to remove the bot from the team and to add it back:

>     a. Go to the Microsoft Teams app.

Reviewer: ```suggestion

    a. Go to the Microsoft Teams app.
```

The integration instance name, `teams` in this example, needs to be configured in the [Configure Microsoft Teams on Cortex XSOAR](#configure-microsoft-teams-on-cortex-xsoar) step. Make sure to set the instance name in all lowercase letters and as one word.

> - Note: You can use the `microsoft-teams-create-messaging-endpoint` command to generate the messaging endpoint, based on the server url, the server version and the instance configurations. For more information see -[microsoft-teams-create-messaging-endpoint documentation](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#microsoft-teams-create-messaging-endpoint).

Reviewer: ```suggestion
- Note: You can use the `microsoft-teams-create-messaging-endpoint` command to generate the messaging endpoint, based on the server URL, the server version, and the instance configurations. For more information, see -[microsoft-teams-create-messaging-endpoint documentation](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#microsoft-teams-create-messaging-endpoint).
```

6. Click **Review + Create**, and wait for the validation to pass.
7. Click **create** if the validation has passed, and wait for the deployment to finish.
8. Under Next Steps, click **Go to resource**.

> 9. Navigate to **Configuration** on the left bar, and fill in the **Messaging Endpoint** (to get the correct messaging endpoint based on the server url, the server version and the instance configurations use the `microsoft-teams-create-messaging-endpoint`command).

Reviewer: ```suggestion
9. Navigate to **Configuration** on the left bar, and fill in the **Messaging Endpoint**. (To get the correct messaging endpoint based on the server URL, the server version, and the instance configurations. use the `microsoft-teams-create-messaging-endpoint`command).
```

There is no context output for this command.

### microsoft-teams-token-permissions-list
***
Retrieves the API permissions associated with the used graph access token. 

> Use this command if you encounter insufficient permissions error when attempting to execute an integration command. Compare the permissions list obtained for the token with the permissions required for the desired command (can be found in the integration documentation), if there are missing API permissions, add them to your application.

Reviewer 2: In case of missing permissions, they will need to reset auth - maybe we should mention this also here https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#microsoft-teams-auth-reset

Author: Thanks! added both here and in the YML file.

There is no context output for this command.

### microsoft-teams-token-permissions-list
***
Retrieves the API permissions associated with the used graph access token. 

> Use this command if you encounter insufficient permissions error when attempting to execute an integration command. Compare the permissions list obtained for the token with the permissions required for the desired command (can be found in the integration documentation), if there are missing API permissions, add them to your application, and then run the `microsoft-teams-auth-reset` command (as described here - [microsoft-teams-auth-reset docs](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#microsoft-teams-auth-reset)).

Reviewer: ```suggestion
Use this command if you encounter insufficient permissions error when attempting to execute an integration command. Compare the permissions list obtained for the token with the permissions required for the desired command (can be found in the integration documentation). If there are missing API permissions, add them to your application, and then run the `microsoft-teams-auth-reset` command (as described here - [microsoft-teams-auth-reset docs](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#microsoft-teams-auth-reset)).
```

##### Base Command

`microsoft-teams-token-permissions-list`

##### Input
There are no input arguments for this command.

##### Context Output
There is no context output for this command.

##### Command Example
```!microsoft-teams-token-permissions-list```

##### Human Readable Output
>### The API permissions obtained for the used graph access token are:
>| Permission          |
>|---------------------|
>| Group.ReadWrite.All |
>| User.Read.All       |
>| Channel.Create      |

### microsoft-teams-create-messaging-endpoint
***

> Generates the messaging endpoint, based on the server url, the server version and the instance configurations. 

Reviewer: ```suggestion
Generates the messaging endpoint, based on the server URL, the server version, and the instance configurations. 
```

> The messaging endpoint should be added to the Demisto bot configuration in Microsoft Teams as part of the Prerequisites of the integration's set-up. For more information see - [Integration Documentation](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#create-the-demisto-bot-in-microsoft-teams).

Reviewer: ```suggestion
The messaging endpoint should be added to the Demisto bot configuration in Microsoft Teams as part of the prerequisites of the integration's setup. For more information see - [Integration Documentation](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#create-the-demisto-bot-in-microsoft-teams).
```

##### Base Command

`microsoft-teams-create-messaging-endpoint`

##### Input

| **Argument Name**   | **Description**                                                                 | **Required** |
|---------------------|---------------------------------------------------------------------------------|--------------|

> | engine_url          | If your instance configuration involve an xsoar engine, provide the engine's IP (or DNS name) and the port in use in the following format - `https://IP:port` or `http://IP:port`. For example - `https://my-engine.name:443`, `http://1.1.1.1:443`. | Optional     | 

Reviewer: ```suggestion
| engine_url          | If your instance configuration involves a Cortex XSOAR engine, provide the engine's IP (or DNS name) and the port in use in the following format - `https://IP:port` or `http://IP:port`. For example - `https://my-engine.name:443`, `http://1.1.1.1:443`. | Optional     | 
```

##### Context Output
There is no context output for this command.

##### Command Example
```!microsoft-teams-create-messaging-endpoint```

##### Human Readable Output
>### The messaging endpoint is: 
>|```https://ext-viso-test.crtx-qa-uat.us.paloaltonetworks.com/xsoar/instance/execute/teams-instance```
>

> > The messaging endpoint should be added to the Demisto bot configuration in Microsoft Teams as part of the Prerequisites of the integration's set up.

Reviewer: ```suggestion
> The messaging endpoint should be added to the Demisto bot configuration in Microsoft Teams as part of the prerequisites of the integration's setup.
```

    This probably means that there is a connection issue, and the web server does not intercept the HTTPS queries from Microsoft Teams.

    To troubleshoot:

>    1. Verify that the messaging endpoint is configured correctly according to the method you chose in the [Setup Examples](#setup-examples) step. If the configuration method you have chosen is rerouting, use the `microsoft-teams-create-messaging-endpoint`command ([microsoft-teams-create-messaging-endpoint documentation](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#microsoft-teams-create-messaging-endpoint)) to get the correct messaging endpoint based on the server url, the server version and the instance configurations.

Reviewer: ```suggestion
   1. Verify that the messaging endpoint is configured correctly according to the method you chose in the [Setup Examples](#setup-examples) step. If the configuration method you have chosen is rerouting, use the `microsoft-teams-create-messaging-endpoint`command ([microsoft-teams-create-messaging-endpoint documentation](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#microsoft-teams-create-messaging-endpoint)) to get the correct messaging endpoint based on the server URL, the server version, and the instance configurations.
```

   6. In some cases, a connection is not created between Teams and the messaging endpoint when adding a bot to the team. You can work around this problem by adding any member to the team the bot was added to (the bot should be already added to the team). This will trigger a connection and solve the issue. You can then remove the member that was added.

> 2. If you see the following error message: `Error in API call to Microsoft Teams: [403] - UnknownError`, then it means the AAD application has insufficient permissions. 

Reviewer: ```suggestion
2. If you see the following error message: `Error in API call to Microsoft Teams: [403] - UnknownError`, it means the AAD application has insufficient permissions. 
```

To retrieves the API permissions associated with the used graph access token you can run the `microsoft-teams-token-permissions-list` command ([microsoft-teams-token-permissions-list documentation](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#microsoft-teams-token-permissions-list)).

> Compare the permissions list obtained for the token with the permissions required for the command you wish to execute (can be found in the command documentation), if there are missing API permissions, add them to your application, and then run the `microsoft-teams-auth-reset` command (as described here - [microsoft-teams-auth-reset documentation](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#microsoft-teams-auth-reset)).

Reviewer: ```suggestion
Compare the permissions list obtained for the token with the permissions required for the command you wish to execute (can be found in the command documentation). If there are missing API permissions, add them to your application, and then run the `microsoft-teams-auth-reset` command (as described here - [microsoft-teams-auth-reset documentation](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#microsoft-teams-auth-reset)).
```
